quarkus-solace-client/runtime/src/main/java/com/solace/quarkus/runtime/OidcProvider.java

```diff
@@ -0,0 +1,59 @@
+package com.solace.quarkus.runtime;
+
+import java.time.Duration;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import com.solace.messaging.MessagingService;
+import com.solace.messaging.config.SolaceProperties;
+
+import io.quarkus.oidc.client.OidcClient;
+import io.quarkus.oidc.client.Tokens;
+import io.quarkus.runtime.StartupEvent;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.infrastructure.Infrastructure;
+
+@ApplicationScoped
+public class OidcProvider {
+
+    @ConfigProperty(name = "quarkus.solace.oidc.refresh.interval", defaultValue = "60s")
+    Duration duration;
+
+    @Inject
+    OidcClient client;
+
+    private volatile Tokens lastToken;
+    private MessagingService service;
+
+    Tokens getToken() {
+        Tokens firstToken = client.getTokens().await().indefinitely();
+        lastToken = firstToken;
+        return firstToken;
+    }
+
+    void init(MessagingService service) {
+        this.service = service;
+    }
+
+    void startup(@Observes StartupEvent event) {
+        Multi.createFrom().ticks().every(duration)
+                .emitOn(Infrastructure.getDefaultWorkerPool())
+                //                .filter(aLong -> {
```

Comment:
Commented it as we will not receive refresh token in client_credentials OAuth flow